src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -401,6 +426,47 @@ public static void NoFormatterFromNegotiation(this ILogger logger, IList<MediaTy
             _noFormatterFromNegotiation(logger, acceptTypes, null);
         }
 
+        public static void InputFormatterSelected(
+           this ILogger logger,
+           IInputFormatter inputFormatter,
+           InputFormatterContext formatterContext)
+        {
+            if (logger.IsEnabled(LogLevel.Debug))
+            {
+                var contentType = formatterContext.HttpContext.Request.ContentType;
+                _inputFormatterSelected(logger, inputFormatter, contentType, null);
+            }
+        }
+
+        public static void InputFormatterRejected(
+            this ILogger logger,
+            IInputFormatter inputFormatter,
+            InputFormatterContext formatterContext)
+        {
+            if (logger.IsEnabled(LogLevel.Debug))
+            {
+                var contentType = formatterContext.HttpContext.Request.ContentType;
+                _inputFormatterRejected(logger, inputFormatter, contentType, null);
+            }
+        }
+
+        public static void NoInputFormatterSelected(
+            this ILogger logger,
+            InputFormatterContext formatterContext)
+        {
+            if (logger.IsEnabled(LogLevel.Debug))
+            {
+                var contentType = formatterContext.HttpContext.Request.ContentType;
+                _noInputFormatterSelected(logger, contentType, null);
+                if (formatterContext.HttpContext.Request.HasFormContentType)
+                {
+                    var modelType = Convert.ToString(formatterContext.ModelType);
```

Comment:
`formatterContext.ModelType.FullName` (save `Convert` for very special occasions).